From 8bfac3685f84f93ae3ac021f257272bcb7d4e3a8 Mon Sep 17 00:00:00 2001
From: Jack Lin <jack.lin@dniserver.dnish.net>
Date: Thu, 24 Feb 2011 18:04:11 +0800
Subject: [PATCH 120/137] netfilter: add a proc entry /proc/sys/dni_netfilter/flush_conntrack_table

Add a proc entry /proc/sys/dni_netfilter/flush_conntrack_table for
flush conntrack table.

To flush current conntrack table:
    echo 1 > /proc/sys/dni_netfilter/flush_conntrack_table

DNI Common Module Patch Review Checklist
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Here are some basic things that I (Jonathan Chang) have checked
in order to ensure the quality of common modules.

1: The syntax and comment are compliant.

2: No memory leak.
   * memory allocated by malloc() will be freed
   * memory allocated by calloc() will be freed

3: No deadlock such as
   * No function call deadlock.
   * No semaphore deadlock.
   * No tag or label deadlock.

4: No stack overflow caused by
   * infinite recursion
   * too large local array variables

5: coding completeness, such as
   * conditions completeness
   * functions completeness
   * update completeness
   * memory resources completeness
   * sessions resources completeness
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Jack Lin <jack.lin@dniserver.dnish.net>
Signed-off-by: Jonathan Chang <jonathan@dniserver.dnish.net>
---
 net/netfilter/dni_netfilter_sysctl.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/net/netfilter/dni_netfilter_sysctl.c b/net/netfilter/dni_netfilter_sysctl.c
index ab14411..584133b 100644
--- a/net/netfilter/dni_netfilter_sysctl.c
+++ b/net/netfilter/dni_netfilter_sysctl.c
@@ -50,6 +50,14 @@ ctl_table dni_table[] = {
 		.proc_handler	= proc_dointvec,
 		.strategy	= sysctl_intvec,
 	},
+	{
+		.ctl_name	= CTL_UNNUMBERED,
+		.procname	= "flush_conntrack_table",
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= proc_flush_conntrack,
+		.strategy	= sysctl_intvec,
+	},
 	{0}
 };
 
-- 
1.7.4.1

